### PR TITLE
feat: add support for destroy_scheduled_duration

### PR DESCRIPTION
**Pull request by chrisw-tracer**
We had a requirement to override the default 24 hour deletion schedule period. I know this default is changing to 30 days next year but thought this might be useful to others in the mean time.

### CODE REVIEW

**Comment by chrisw-tracer** [comment]
> Thanks for your pull request! It looks like this may be your first contribution to a Google open source project. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> View this [failed invocation](https://github.com/terraform-google-modules/terraform-google-kms/pull/109/checks?check_run_id=18273970862) of the CLA check for more information.
> 
> For the most up to date status, view the checks section at the bottom of the pull request.

I've completed the CLA

**Comment by chrisw-tracer** [comment]
@apeabody any feedback on this PR?

File: variables.tf [hunk 1]
```diff
@@ -42,6 +42,12 @@ variable "prevent_destroy" {
   default     = true
 }
 
+variable "key_destroy_scheduled_duration" {
+  description = "Set The period of time that versions of keys spend in the DESTROY_SCHEDULED state before transitioning to DESTROYED."
```
Comment (apeabody) [inline variables.tf:46]: nit: typo

```suggestion
  description = "Set the period of time that versions of keys spend in the DESTROY_SCHEDULED state before transitioning to DESTROYED."
```

File: variables.tf [hunk 2]
```diff
@@ -42,6 +42,12 @@ variable "prevent_destroy" {
...
+  type        = string
+  default     = "86400s"
```
Comment (apeabody) [inline variables.tf:48]: Hmm, you mention an upcoming change to the API's default.  Would it work to use `null` here instead to allow for that change?  If the API instead returns an actual value such as `86400s`, that would unfortunately cause config churn on subsequent applies.

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks for the contribution @chrisw-tracer!

File: variables.tf (hunk 2 above)
Comment (chrisw-tracer) [reply variables.tf:48]: That is a great point, didn't think of that. I'll update accordingly!

File: variables.tf (hunk 1 above)
Comment (chrisw-tracer) [reply variables.tf]: fixed

File: variables.tf (hunk 2 above)
Comment (chrisw-tracer) [reply variables.tf]: and fixed

**Review by apeabody — APPROVED** [review]

**Pull request merged**